### ISSUE TITLE
Protected readonly field reporting SA1401

### ISSUE THREAD

**Issue by GregReddick** [DotNetAnalyzers/StyleCopAnalyzers#1779]
First reported in #1766, Issue D10, with some refinements here.

The following code is reporting SA1401. Classic StyleCop did not report anything.

```
namespace RuleViolations
{
    /// <summary>A rule SA1401 fail.</summary>
    public class RuleSA1401Fail
    {
        /// <summary>The value.</summary>
        protected readonly int Value = 1;
    }
}
```

**Comment by pdelvo** [comment]
Whatever we decide to do here we should change the documentation to reflect that. If I follow the documentation reporting SA1401 is the right thing to do.

**Comment by mnivet** [comment]
The documentation already explicitly says that readonly fields are excluded from the rule:

> Fields that are static and readonly will not raise a violation. These kinds of fields are commonly used to represent a constant value when the const keyword cannot be used, and therefore they are exempt from this rule.

In my point of view it's an issue in Analyzer implementation.

**Comment by sharwell** [comment]
@mnivet This issue relates to fields which are `static` but *not* `readonly`. The documentation you mentioned relates to fields which are both `static` and `readonly`.

**Comment by mnivet** [comment]
@sharwell ok, "I've understand it like Fields that are `static` and (fields that are) `readonly`".
But I've read it again today and found that it was not logical with the explained use case of constant values.

**Comment by MarcAndreSG** [comment]
Hey all,

I do not understand the meaning in it.
If I mark a member variable as protected,
I want them to be visible in the derived class.
I do not always want a property that can set or get the value.
Why must language features with any rules so meaningless
remove?

Marcus

**Comment by sharwell** [comment]
@MarcAndreSG It's fine to not agree with a rule. Just disable it. 😄
